### ISSUE TITLE
Can't serialize with trie.T as property is private 

### ISSUE THREAD

**Issue by Bishoy** [sidvishnoi/compact-prefix-tree#1]
I might be wrong but since T is a private property of `CompactPrefixTree`,  I'm not sure how the sample would work serializing `JSON.stringify(trie.T)` and then retrieving the trie back . I would suggest instead something along the lines of : 

```
const items = [
  "http://www.example.com/foo/",
  "https://www.example.com/baz/",
];
const trie = new CompactPrefixTree(items);
const serialized = JSON.stringify(trie);
 
// to deserialize
const trie2 = Object.assign(new CompactPrefixTree(),JSON.parse(serialized))
```

**Comment by sidvishnoi** [comment]
You're right on `T` being private. I probably ignored TS errors when writing that example.

I think it would be better to add a `toJSON()` method to `CompactPrefixTree` as well as a `static fromJSON(json)` constructor. Interested in sending a PR?
